src/NHibernate.Test/NHSpecificTest/NH1001/Fixture.cs

```diff
@@ -207,19 +205,44 @@ public void Department2IsNull_QueryOver()
 		{
 			ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 99999 WHERE EMPLOYEE_ID = {employeeId}");
 
-			IStatistics statistics = Sfi.Statistics;
+			var statistics = Sfi.Statistics;
 			statistics.Clear();
 
-			using (ISession session = OpenSession())
+			using (var session = OpenSession())
 			{
 				var employee = session.QueryOver<Employee>()
 					.Fetch(SelectMode.Fetch, e => e.Department2)
 					.Fetch(SelectMode.Skip, e => e.Department1, e => e.Department3, e => e.Address)
 					.Where(e => e.Id == employeeId)
 					.SingleOrDefault();
+				Assert.That(employee.Department2, Is.Null);
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(4), "Employee, Department1, Department3, and Address");
 
 				Assert.That(employee.Department1, Is.Not.Null);
+				Assert.That(employee.Department3, Is.Not.Null);
+				Assert.That(employee.Address, Is.Not.Null);
+			}
+		}
+
+		[Test]
+		public void Department2IsNullSkipPhones_QueryOver()
+		{
+			ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 99999 WHERE EMPLOYEE_ID = {employeeId}");
+
+			var statistics = Sfi.Statistics;
+			statistics.Clear();
+
+			using (var session = OpenSession())
+			{
+				var employee = session.QueryOver<Employee>()
+					.Fetch(SelectMode.Fetch, e => e.Department2)
+					.Fetch(SelectMode.Skip, e => e.Department1, e => e.Department3, e => e.Address, e => e.Phones)
+					.Where(e => e.Id == employeeId)
+					.SingleOrDefault();
 				Assert.That(employee.Department2, Is.Null);
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(5), "Employee, Department1, Department3, Address, and Phones");
```

Comment:
Are you sure regarding Phones? Additional query is required only for "single property" mappings (like many-to-one)  - because `not-found` property needs to be initialized with null in case not found case.  `Phones` shouldn't trigger "not found" check statement.  That's something I was trying to discuss [here](https://github.com/nhibernate/nhibernate-core/pull/1948#discussion_r249247393)

So I would say it should be 4

Reply:
Yes, I'm sure about phones. It is mapped not lazily.

Reply:
The queries generated (except for `Department2`) not because of "not found", but because they are not lazy.

Reply:
No.. `SelecMode.Skip` overrides mapping. That's the whole point of using it. You've made `Phones` lazy in query.

Reply:
These are the queries generated
```sql
-- 1
    SELECT
        this_.EMPLOYEE_ID as employee1_3_1_,
        this_.FIRST_NAME as first2_3_1_,
        this_.LAST_NAME as last3_3_1_,
        this_.DEPARTMENT_ID_1 as department4_3_1_,
        this_.DEPARTMENT_ID_2 as department5_3_1_,
        this_.DEPARTMENT_ID_3 as department6_3_1_,
        this_.ADDRESS_ID as address7_3_1_,
        department2_.DEPARTMENT_ID as department1_0_0_,
        department2_.DEPARTMENT_NAME as department2_0_0_ 
    FROM
        EMPLOYEES this_ 
    left outer join
        DEPARTMENTS department2_ 
            on this_.DEPARTMENT_ID_2=department2_.DEPARTMENT_ID 
    WHERE
        this_.EMPLOYEE_ID = @p0;
    @p0 = 1 [Type: Int32 (0:0:0)]
--2 Department1
    SELECT
        department0_.DEPARTMENT_ID as department1_0_0_,
        department0_.DEPARTMENT_NAME as department2_0_0_ 
    FROM
        DEPARTMENTS department0_ 
    WHERE
        department0_.DEPARTMENT_ID=@p0;
    @p0 = 11 [Type: Int32 (0:0:0)]
-- 3 Department2 - unwanted 
    SELECT
        department0_.DEPARTMENT_ID as department1_0_0_,
        department0_.DEPARTMENT_NAME as department2_0_0_ 
    FROM
        DEPARTMENTS department0_ 
    WHERE
        department0_.DEPARTMENT_ID=@p0;
    @p0 = 99999 [Type: Int32 (0:0:0)]
-- 4 Department 3 
    SELECT
        department0_.DEPARTMENT_ID as department1_0_0_,
        department0_.DEPARTMENT_NAME as department2_0_0_ 
    FROM
        DEPARTMENTS department0_ 
    WHERE
        department0_.DEPARTMENT_ID=@p0;
    @p0 = 13 [Type: Int32 (0:0:0)]
-- 5 Address  
    SELECT
        addressemp0_.ADDRESS_ID as address1_1_0_,
        addressemp0_.ADDRESS_NAME as address2_1_0_ 
    FROM
        ADDRESSES addressemp0_ 
    WHERE
        addressemp0_.ADDRESS_ID=@p0;
    @p0 = 15 [Type: Int32 (0:0:0)]
-- 6 Phones 
    SELECT
        phones0_.EMPLOYEE_ID as employee3_2_1_,
        phones0_.PHONE_ID as phone1_2_1_,
        phones0_.PHONE_ID as phone1_2_0_,
        phones0_.PHONE_NUMBER as phone2_2_0_ 
    FROM
        PHONES phones0_ 
    WHERE
        phones0_.EMPLOYEE_ID=@p0;
    @p0 = 1 [Type: Int32 (0:0:0)]
```

If phones is not skipped then last query is not generated but embedded in the first query.

Reply:
Well. That doesn't look right to me. Maybe something in mapping (like lazy="false" for Employee) triggers such behavior or it's some issue with SelecMode.Skip. But it's off topic.

Reply:
Ah.. Indeed all classes and `Phones` property is mapped with `lazy="false"`. 
But `SkipMode` overrides only fetch mode (which is `fetch="join"` in our case).